contracts/src/v0.8/dev/VRFCoordinatorV2.sol

```diff
@@ -0,0 +1,527 @@
+    if (staleFallback && stalenessSeconds < block.timestamp - timestamp) {
+      linkEth = s_fallbackLinkPrice;
+    }
+    return uint256(linkEth);
+  }
+
+  function withdraw(
+    address recipient,
+    uint96 amount
+  )
+    external
+  {
+    if (s_withdrawableTokens[msg.sender] < amount) {
+      revert InsufficientBalance();
+    }
+    s_withdrawableTokens[msg.sender] -= amount;
+    assert(LINK.transfer(recipient, amount));
+  }
+
+  function getSubscription(
+    uint64 subId
+  )
+    external
+    view
+    returns (
+      Subscription memory
+    )
+  {
+    return s_subscriptions[subId];
+  }
+
+  function createSubscription(
+    address[] memory consumers // permitted consumers of the subscription
+  )
+    external
+    validConsumers(consumers)
+    returns (
+      uint64
+    )
+  {
+    currentSubId++;
+    s_subscriptions[currentSubId] = Subscription({
+      owner: msg.sender,
+      consumers: consumers,
+      balance: 0
+    });
+    emit SubscriptionCreated(currentSubId, msg.sender, consumers);
+    return currentSubId;
+  }
+
+  function updateSubscription(
+    uint64 subId,
+    address[] memory consumers // permitted consumers of the subscription
+  )
+    external
+    onlySubOwner(subId)
+    validConsumers(consumers)
+  {
+    address[] memory oldConsumers = s_subscriptions[subId].consumers;
+    s_subscriptions[subId].consumers = consumers;
+    emit SubscriptionConsumersUpdated(subId, oldConsumers, consumers);
+  }
+
+  function fundSubscription(
+    uint64 subId,
+    uint96 amount
+  )
+    external
+    onlySubOwner(subId)
+  {
+    if (s_subscriptions[subId].owner == address(0))  {
+      revert InvalidSubscription();
+    }
+    uint256 oldBalance = s_subscriptions[subId].balance;
+    s_subscriptions[subId].balance += amount;
+    LINK.transferFrom(msg.sender, address(this), amount);
+    emit SubscriptionFundsAdded(subId, oldBalance, s_subscriptions[subId].balance);
+  }
+
+  function withdrawFromSubscription(
+    uint64 subId,
+    address to,
+    uint96 amount
+  )
+    external
+    onlySubOwner(subId)
+  {
+    if (s_subscriptions[subId].balance < amount) {
+      revert InsufficientBalance();
+    }
+    uint256 oldBalance = s_subscriptions[subId].balance;
+    s_subscriptions[subId].balance -= amount;
+    LINK.transfer(to, amount);
+    emit SubscriptionFundsWithdrawn(subId, oldBalance, s_subscriptions[subId].balance);
+  }
+
+  // Keep this separate from zeroing, perhaps there is a use case where consumers
+  // want to keep the subId, but withdraw all the link.
+  function cancelSubscription(
+    uint64 subId,
+    address to
+  )
+    external
+    onlySubOwner(subId)
+  {
+    uint256 balance = s_subscriptions[subId].balance;
+    delete s_subscriptions[subId];
+    LINK.transfer(to, balance);
+    emit SubscriptionCanceled(subId, to, balance);
+  }
+
+  modifier onlySubOwner(uint64 subId) {
+    if (msg.sender != s_subscriptions[subId].owner) {
+      revert MustBeSubOwner();
```

Comment:
It might make things a little easier for users if `MustBeSubOwner` contained the owner address.